packages/protocol/test/tokenomics/utils.ts

```diff
@@ -1,6 +1,48 @@
 import { BigNumber, ethers } from "ethers";
 import { TaikoL1, TkoToken } from "../../typechain";
+import deployAddressManager from "../utils/addressManager";
 import Proposer from "../utils/proposer";
+import {
+    getDefaultL2Signer,
+    getL1Provider,
+    getL2Provider,
+} from "../utils/provider";
+import createAndSeedWallets from "../utils/seed";
+import sleep from "../utils/sleep";
+import { defaultFeeBase, deployTaikoL1 } from "../utils/taikoL1";
+import { deployTaikoL2 } from "../utils/taikoL2";
+import deployTkoToken from "../utils/tkoToken";
+import { ethers as hardhatEthers } from "hardhat";
+import { BlockProposedEvent } from "../../typechain/LibProposing";
+import { expect } from "chai";
+import verifyBlocks from "../utils/verify";
+
+type ForkChoice = {
+    provenAt: BigNumber;
+    provers: string[];
+    blockHash: string;
+};
+
+type BlockInfo = {
+    proposedAt: number;
+    provenAt: number;
+    id: number;
+    parentHash: string;
+    blockHash: string;
+    forkChoice: ForkChoice;
+    deposit: BigNumber;
+    proposer: string;
+};
+
+async function sleepUntilBlockIsVerifiable(
+    taikoL1: TaikoL1,
+    id: number,
+    provenAt: number
+) {
+    const delay = await taikoL1.getUncleProofDelay(id);
+    const delayInMs = delay.mul(1000);
+    await sleep(5 * delayInMs.toNumber()); // TODO: use provenAt, calc difference, etc
```

Comment:
L1 is a geth node, not hardhat.
Maybe we should change both to be hardhat nodes.

Reply:
https://github.com/NomicFoundation/hardhat/issues/3345 I'm not sure but changing both to hardhat may affect some of our other tests? I remember having trouble with this when I was testing some other stuff

Reply:
We can change the image for the Tokenomics tests vs the Integration tests which dont require eth_getProof.

Reply:
@dantaik @davidtaikocha is this soemthign we want to do? change nodes?

Reply:
sounds good, but looks like this feature request's status is BLOCKED?
![image](https://user-images.githubusercontent.com/104078303/215929118-461fdd07-a6f6-43ff-a9c6-42d8af079013.png)

Reply:
No, I mean change the tokenomics tests (not rest of integration tests) to use two hardhat modes instead if own geth node,. Since tokenomics doesn't use getProof, we don't have to wait.

Reply:
Sounds good to me, i believe this will make the tests faster and simpler.

Reply:
@davidtaikocha seems multiple hardhat node is not natively supported, there are some workarounds but they are very clunky. There are two open Github Issues regarding peoiple waiting for them to support multiple hardhat nodes concurrently with less issues.

Do you have any idea of a smooth workaround? If we change both to be hardhat nodes, they use the same hardhat config, and share chainIds, and doesnt seem to work well. Not sure if its worth the investigation.

Reply:
Not quite sure but since currently our tokenomics tests are started by `test_integration.sh`, does it mean after simply [removing the `--network l1_test` flag here](https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/test/test_integration.sh#L83), we will have two hardhat node in tests? (one started through docker in `test_integration.sh` and another is started by `hardhat test`)